pkg/extra/handlers.go

```diff
@@ -32,8 +41,34 @@ func (a *API) handleConsent(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	consentChallenge := r.URL.Query().Get("consent_challenge")
```

Comment:
question: are we good even if the value is empty here? or should we add the check and behave differently?